osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -105,9 +108,22 @@ private void load(AudioManager audio, DialogOverlay dialog, OsuColour colours, S
             // initial value transfer is required for FilterControl (it uses our re-cached bindables in its async load for the initial filter).
             transferRulesetValue();
 
+            LoadComponentAsync(Carousel = new BeatmapCarousel
+            {
+                AllowSelection = false, // delay any selection until our bindables are ready to make a good choice.
+                Anchor = Anchor.CentreRight,
+                Origin = Anchor.CentreRight,
+                RelativeSizeAxes = Axes.Both,
+                BleedTop = FilterControl.HEIGHT,
+                BleedBottom = Footer.HEIGHT,
+                SelectionChanged = updateSelectedBeatmap,
+                BeatmapSetsChanged = carouselBeatmapsLoaded,
+                GetRecommendedBeatmap = (recommender = new DifficultyRecommender()).GetRecommendedBeatmap,
```

Comment:
Bit of a nit: How about moving this construction to the field, and making the field `readonly`? Quite easy to miss that's it's being done in a nested invocation, and then added later on.

Reply:
sounds good